### ISSUE TITLE
[FEATURE] Sorting with folders + Blacklist

### ISSUE THREAD

**Issue by Alsonn** [HDoujinDownloader/HDoujinDownloader#406]
I found out how to set specific main folders for artists, but I was wondering if there could be a way to sort with folders, inside the list of manga in the downloader itself ? Just to see a bit more clearly ? 

+

I think it would be interesting to have some kind of a blacklist ? 
For example, I just added all the albums from one artist, but in the downloader I found out that there is an album that is a compilation of different artists and I don't want to download it. 
An "add to blacklist" option would be helpful to mark this specific album as "not to download" ?

Thanks for the good work, I'm still figuring out the possibilities of your software before buying it, but for now it's really impressive !

**Comment by Alsonn** [comment]
Yeah, I just bought a license because it was so great. But now I have a lot of albums in my download tab and it's a bit laggy. Maybe a folder system would help that too ? I don't want to remove them in order to keep some kind of database. I tried using the history feature but I had some error messages and could'nt download anymore. Would this jeopardise the stability of the software ?

**Comment by HDoujinDownloader** [comment]
> I tried using the history feature but I had some error messages and could'nt download anymore.

The history feature is definitely the best way of keeping track of what you've downloaded. It even stores the album metadata to facilitate searching, which can be useful. Can I ask what errors were appearing? It shouldn't affect stability, but it's possible you've encountered a bug.

There are definitely some things I could do to optimize how the download queue is accessed, but for the time being, yes, extremely large queues are likely to introduce some slowdown.

> I just bought a license because it was so great.

Thank you so much for the support, by the way! ❤️

**Comment by Alsonn** [comment]
I couldn't make happen the history error for now but I will let you know when it does :)

To give you an example of my usage : I want to be able to download all of the albums of all the artists that I'm following. An automatic full scrape of all the pages would be the best but in the mean time this is what I do : 

With the usage that I have I use the "generate url" to get all the pages for one artist but it can get a bit tedious doing this one by one with the "?page=(1-X)" to add to an artist link. I know that you can load a list (but the list would very long and not adaptative if new pages are added). So I thought that maybe there could be a way to load a list and through an automatic edit add "?page=(1-X)" (defined by the user somewhere else) to all of the lines ?

**Comment by HDoujinDownloader** [comment]
I have a possibly better solution for you: Go to **_Tools > General settings > Advanced settings_**, search for the `Downloads.MaxScrapingDepth` setting, and set it to something high to represent the maximum number of pages you want it to add galleries from. You can now avoid having to generate the URLs, and it will automatically scrape the number of pages specified.

This isn't implemented for all sites yet, so if it's not working for the one you're downloading from, let me know and I'll upload a fix accordingly.

**Comment by Alsonn** [comment]
Yeah I already explored this setting I've set a 2000 limit (a bit much I admit) but it doesn't seem to work. I mainly download from nhentai.net and imhentai.xxx (or comicporn.xxx don't know wich one is best).

**Comment by HDoujinDownloader** [comment]
The setting was implemented for nhentai.net, but not imhentai.xxx/comicporn.xxx. I just uploaded a fix, so it should be working on all of them now. Let me know if you have any problems!

**Comment by Alsonn** [comment]
Thanks a lot ! Seems to be working perfectly :)

**Comment by Alsonn** [comment]
Hi again, I was wondering if there could be a feature when the software automatically detect the name of albums (basically ignoring what is in between brackets) and then group identical/similar together ? Like I download from imhentai and nhentai to cross data, and it would help in order to then choosing witch language I prefer for one manga/album.
It could maybe use the similarity of name or hash to find duplicates. I know it already find album with identical name duplicates.

Also how does the software works if an albums is updated ? Does it update the one previsously downloaded or redownload it like a brand new album ?

It does look like the deep scraping is not implemented for hitomi.la, could it be implemented ?
